l10n_ch_scan_bvr/wizard/scan_bvr.py

```diff
@@ -485,61 +348,64 @@ def validate_bvr_string(self, cr, uid, ids, context):
         # Explode and check  the BVR Number and structurate it
         #
         data = {}
-        data['bvr_struct'] = self._get_bvr_structurated(
-            bvr_data.bvr_string)
+        data['bvr_struct'] = self._get_bvr_structurated(self.bvr_string)
+        partner_bank_model = self.env['res.partner.bank']
+        partner_bank = False
         # We will now search the account linked with this BVR
         if data['bvr_struct']['domain'] == 'name':
             domain = [('acc_number', '=', data['bvr_struct']['beneficiaire'])]
-            partner_bank_search = self.pool.get('res.partner.bank').search(
-                cr,
-                uid,
-                domain,
-                context=context
-            )
+            partners_bank = partner_bank_model.search(domain)
+            partner_bank = len(partners_bank) == 1 and partners_bank[0]
         else:
-            domain = [
-                ('bvr_adherent_num', '=', data['bvr_struct']['bvrnumber'])
-            ]
-            partner_bank_search = self.pool.get('res.partner.bank').search(
-                cr,
-                uid,
-                domain,
-                context=context
-            )
+            # A postal account that refers to a bank is uniquely identified
+            # by (beneficiaire + bvrnumber), at least by beneficiaire
+            domain = [('acc_number', '=', data['bvr_struct']['beneficiaire'])]
+            partners_bank = partner_bank_model.search(domain)
+            if len(partners_bank) == 1:
+                if (
+                        not partners_bank[0].bvr_adherent_num or
+                        partners_bank[0].bvr_adherent_num ==
+                        data['bvr_struct']['bvrnumber']):
+                    partner_bank = partners_bank
+            elif len(partners_bank) > 1:
```

Comment:
As stated previously, unless the sql_constraint has been dropped somehow, there can be only one in version 9.0.
https://github.com/odoo/odoo/blob/9.0/openerp/addons/base/res/res_bank.py#L70